app/services/course/assessment/question/programming/language_package_service.rb

```diff
@@ -60,14 +68,37 @@ def extract_meta(attachment, template_files) # rubocop:disable UnusedMethodArgum
     raise NotImplementedError, 'You must implement this'
   end
 
+  # Retrieves the absolute path of the file specified
+  #
+  # @param [String] curr_dir The current directory of the specific language package service
+  # @param [String] filename The filename of the file to get the path of
+  def self.get_file_path(curr_dir, filename)
+    File.join(File.expand_path(File.dirname(__FILE__)), curr_dir, filename).freeze
```

Comment:
If I am not wrong, I think there is no need to specify `curr_dir`, replace it with `self` and it should work. The self in this case evaluates to the context of the caller, which would be the specific language service. 

Can try out and verify?